otwcode/otwarchive
# AO3-6868 Add language form validation

# Pull Request Checklist

  as the *first* thing in your pull request title (e.g. `AO3-1234 Fix thing`)
  until they are reviewed and merged before creating new pull requests.

## Issue

https://otwarchive.atlassian.net/browse/AO3-6868

## Purpose

This adds two form validations to the language form on the `/languages/<language abbreviation/edit` page:
* validates that the display name is not a duplicate
* validates that the abbreviation is 4 characters or fewer 

Also:
* added rake task to check for duplicate language names before running migrations
    * `rake db:check_language_name_duplicates`
* fixed some existing tests that were failing on the new validations and rubocop

## Testing Instructions

1. Login as an admin
2. Go to `/languages`
3. Click `Edit` on any of the languages
4. Change the display name to one that already exists
5. Change the abbreviation to something with more than 5 characters
6. Click `Update Language` and form validations should appear

<img width="574" alt="Screenshot 2024-12-27 at 10 37 58 AM" src="https://github.com/user-attachments/assets/1e44753b-e911-4fba-b232-e4746311b7d9" />

## Credit

Connie Feng, she/her
([JIRA account](https://otwarchive.atlassian.net/jira/people/712020:17930fa1-d647-47bd-b334-8a84cbacfca7))
